app/models/generic_object.rb

```diff
@@ -95,6 +103,28 @@ def property_setter(name, value)
     self.properties = properties.merge(name => val)
   end
 
-  def call_automate(method_name, options = {}, q_options = {})
+  # the method parameters are passed into automate as a hash: {:param_1 => param_1, :param_2 => param_2}
+  # the return value from automate is in $evm.root['method_result']
+  def call_automate(method_name, *args)
+    raise "A user is required to send [#{method_name}] to automate." unless @user
+
+    attrs = {
+      :go_definition_name => generic_object_definition.name,
+      :method_name        => method_name
+    }
+    args.each_with_index { |item, idx| attrs["param_#{idx + 1}".to_sym] = item }
```

Comment:
@mkanoor I know you and @lfu discussed this issue.  What did you end up deciding?